src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.addAnimal(this);
+  }
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
```

Comment:
here you are comparing object to object(reference to reference), either we can't compare their names, because we can have animals with same names.  
Let's rename this method to `removeDeadAnimals`, and filter animals which have health > 0